src/ezlogs_cli.ml

```diff
@@ -1,22 +1,94 @@
-let now_fmt fmt () =
-  let { Unix.tm_year; tm_mon; tm_mday; tm_hour; tm_min; tm_sec; _ } =
-    Unix.gmtime @@ Unix.gettimeofday ()
-  in
-  Fmt.pf fmt "%04d-%02d-%02d %02d:%02d:%02d " (tm_year + 1900) (tm_mon + 1)
-    tm_mday tm_hour tm_min tm_sec
+module Line = struct
+  let now_fmt fmt () =
+    let { Unix.tm_year; tm_mon; tm_mday; tm_hour; tm_min; tm_sec; _ } =
+      Unix.gmtime @@ Unix.gettimeofday ()
+    in
+    Fmt.pf fmt "%04d-%02d-%02d %02d:%02d:%02d " (tm_year + 1900) (tm_mon + 1)
+      tm_mday tm_hour tm_min tm_sec
 
-let pp_header =
-  Fmt.suffix Fmt.(const string " ") @@ Fmt.prefix now_fmt Logs_fmt.pp_header
+  let pp_header =
+    Fmt.suffix Fmt.(const string " ") @@ Fmt.prefix now_fmt Logs_fmt.pp_header
 
-let setup_log style_renderer level =
-  Fmt_tty.setup_std_outputs ?style_renderer ();
-  Logs.set_level level;
-  Logs.set_reporter (Logs_fmt.reporter ~pp_header ());
-  ()
+  let setup style_renderer level =
+    Fmt_tty.setup_std_outputs ?style_renderer ();
+    Logs.set_level level;
+    Logs.set_reporter (Logs_fmt.reporter ~pp_header ());
+    ()
 
-(* Command line interface *)
+  let logging =
+    Cmdliner.Term.(const setup $ Fmt_cli.style_renderer () $ Logs_cli.level ())
+end
 
-open Cmdliner
+module Json = struct
+  module Json = Yojson.Basic
 
-let setup_log =
-  Term.(const setup_log $ Fmt_cli.style_renderer () $ Logs_cli.level ())
+  let dict_of_tags (tags : Logs.Tag.set) =
+    Logs.Tag.fold
+      (fun tag dict ->
+        match tag with
+        | V (tag_definition, tag_value) ->
+          let name = Logs.Tag.name tag_definition in
+          let tag_string =
+            Fmt.str "%a" (Logs.Tag.printer tag_definition) tag_value
+          in
+          (name, `String tag_string) :: dict)
+      tags []
+
+  let reporter ppf =
+    let report src level ~over k msgf =
+      let continuation _ =
+        over ();
+        k ()
+      in
+      let as_json _header tags k ppf fmt =
+        Fmt.kstr
+          (fun message ->
+            let tags_dict =
+              match tags with
+              | None -> []
+              | Some set -> dict_of_tags set
+            in
+            let json : Json.t =
+              `Assoc
+                (("level", `String (Logs.level_to_string (Some level)))
+                :: ("source", `String (Logs.Src.name src))
+                :: ("message", `String message)
```

Comment:
Please events :) https://www.elastic.co/guide/en/ecs/current/ecs-event.html
I would love to have an event / alert in the log like `BenignerFoundUniqueExample` so I could aggregate over them

Reply:
Maybe also log level in a separate field?

Reply:
The log level is here, unless you meant something different? https://github.com/hcarty/ezlogs-cli/pull/1/files#diff-5ab8778f2f1fd8509c6ec40348d088cbR53

For events, I think we can capture them with [tags](https://erratique.ch/software/logs/doc/Logs/Tag/index.html).  It's worth having convenient support for that but I'm not sure if that should live in this library or somewhere else.  If it lives in this library then we should probably change the name since it's getting away from just CLI parsing + logging setup.

Expanding scope + renaming would not be a bad thing since we haven't made a proper opam release yet.  And we if we make it ECS-ready and APM-ready then that seems well worth expanding into.

Reply:
Moved log level to `log.level` for more ECSing.  Thanks!